src/sql/util.py

```diff
@@ -363,3 +365,123 @@ def show_deprecation_warning():
         "raise an exception in the next major release so please remove it.",
         FutureWarning,
     )
+
+
+def find_path_from_root(file_name):
+    """
+    Recursively finds an absolute path to file_name starting
+    from current to root directory
+    """
+    current = Path().resolve()
+    while not (current / file_name).exists():
+        if current == current.parent:
+            return None
+
+        current = current.parent
+    display.message(f"Found {file_name} from '{current}'")
+
+    return str(Path(current, file_name))
+
+
+def find_close_match_config(word, possibilities, n=3):
+    """Finds closest matching configurations and displays message"""
+    closest_matches = difflib.get_close_matches(word, possibilities, n=n)
+    if not closest_matches:
+        display.message(
+            f"'{word}' is an invalid configuration. Please review our "
+            "configuration guideline: "
+            "https://jupysql.ploomber.io/en/latest/api/configuration.html#options"
+        )
+    else:
+        display.message(
+            f"'{word}' is an invalid configuration. Did you mean "
+            f"{pretty_print(closest_matches, last_delimiter='or')}?"
+        )
+
+
+def get_line_content_from_toml(file_path, line_number):
+    """
+    Locates a line that error occurs when loading a toml file
+    and returns the line, key, and value
+    """
+    with open(file_path, "r") as file:
+        lines = file.readlines()
+        eline = lines[line_number - 1].strip()
+        ekey, evalue = None, None
+        if "=" in eline:
+            ekey, evalue = map(str.strip, eline.split("="))
+        return eline, ekey, evalue
+
+
+def load_toml(file_path):
+    """
+    Returns toml file content in a dictionary format
+    and raises error if it fails to load the toml file
+    """
+    try:
+        with open(file_path, "r") as file:
+            content = file.read()
+            return toml.loads(content)
+    except toml.TomlDecodeError as e:
+        eline, ekey, evalue = get_line_content_from_toml(file_path, e.lineno)
+        if "Duplicate keys!" in str(e):
+            raise ValueError(f"Duplicate key found : '{ekey}'") from None
+        elif "Only all lowercase booleans" in str(e):
+            raise ValueError(
```

Comment:
this can be a separate function maybe `parse_toml_error`. I see this now 
```

File ~/miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/util.py:428, in load_toml(file_path)
    426 eline, ekey, evalue = get_line_content_from_toml(file_path, e.lineno)
    427 if "Duplicate keys!" in str(e):
--> 428     raise ValueError(f"Duplicate key found : '{ekey}'") from None
    429 elif "Only all lowercase booleans" in str(e):
    430     raise ValueError(
    431         f"Invalid value '{evalue}' in '{eline}'. "
    432         "Valid boolean values: true, false"
    433     ) from None

ValueError: Duplicate key found : 'feedback'
```

No need to print stack trace, just the error

Reply:
use UsageError so only the error message is displayed (i.e.g, don't convert errors into print statements)

Reply:
Now using exceptions.ConfigurationError (it's a new one)